include/settings/settings.h

```diff
@@ -111,6 +113,15 @@ int settings_subsys_init(void);
  */
 int settings_register(struct settings_handler *cf);
 
+/**
```

Comment:
If we wish to go towards link time registration in the future I would rather not implement this functionality.

Reply:
I mean - the settings_unregister function implementation is risky.

Reply:
Why do you think settings_unregister() is risky ? The implementation or the functionality ?

Reply:
Because it would make it hard to move towards compile/link time registration. And I really believe that settings module should not be used for anything that has dynamically allocated callback during device lifetime - this seems much more like a task for some filesystem.

Reply:
good point @rakons I don't see a case for unregistered. I see that move to compile time registration make sense.
It is always possible to add handler disabling feature which will does the same in effect.

Reply:
Agreed with @rakons and @nvlsianpu, we should go for compile/link time registration.

Reply:
I also agree with this, but it does not have to be a part of this PR. And an even better approach would be not to do any registration at all
As it is the PR can be used to reduce any RAM waste by deregistering. When a compile/link registration is used (not his PR) the register routiine needs to be replaced with something else and deregister will disappear.

Reply:
Can we move this to separate PR / discussion?

Reply:
maybe this can be optional?

Reply:
Let's decide on this on the meeting.